### PR TITLE
Feature/267 Add rule to replace setMethods to onlyMethods

### PR DESCRIPTION
**Pull request by eerison**
Issue #267

### CODE REVIEW

**Comment by eerison** [comment]
Wait for this PR https://github.com/rectorphp/rector-phpunit/pull/278

File: rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    /**
+     * @param MethodCall $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $this->testsNodeAnalyzer->isPHPUnitMethodCallNames($node, ['setMethods'])) {
+            return null;
+        }
+
```
Comment (samsonasik) [inline rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:66]:
```suggestion

        if ($node->isFirstClassCallable()) {
            return null;
        }
```

File: rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        $args = $node->getArgs();
+
```
Comment (samsonasik) [inline rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:71]:
```suggestion
        if (! isset($node->args[0])) {
            return null;
        }
        
```

File: rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\PHPUnit\PHPUnit90\Rector\MethodCall;
```
Comment (samsonasik) [inline rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:5]: is it phpunit 9 or phpunit 10? If phpunit 10, then it needs in `PHPUnit100` namespace imo:

it also may need to be added to the phpunit 10 setlist

https://github.com/rectorphp/rector-phpunit/blob/main/config/sets/phpunit100.php

Comment (eerison) [reply rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:5]: it was deprecated phpuni9 and removed phpunit10 in this case do we apply for versions that we are removing?

Comment (samsonasik) [reply rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:5]: I see, but it seems still need to be registered to `config/sets/phpunit100.php`

Comment (eerison) [reply rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:5]: then should I move to PhpUnit100 folder and register in `config/sets/phpunit100.php`?

Comment (samsonasik) [reply rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:5]: I think using `PHPUnit` directory is better, but probably @TomasVotruba has opinion?

Comment (eerison) [reply rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:5]: Hey @samsonasik 

I guess I could just use 

it 😄 
```php
new MethodCallRename('PHPUnit\Framework\Assert', 'setMethods', 'onlyMethods'),
```

Comment (eerison) [reply rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:5]: > https://github.com/rectorphp/rector-phpunit/blob/main/config/sets/phpunit100.php

like this https://github.com/rectorphp/rector-phpunit/blob/main/config/sets/phpunit100.php#L58 😄

Comment (samsonasik) [reply rules/PHPUnit90/Rector/MethodCall/UseOnlyMethodRector.php:5]: yeah, that seems good 👍

**Review by samsonasik — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you

**Pull request merged**